src/elements/content-sidebar/hooks/useSidebarMetadataFetcher.ts

```diff
@@ -122,59 +125,52 @@ function useSidebarMetadataFetcher(
         [onApiError],
     );
 
-    const deleteMetadataInstanceSuccessCallback = React.useCallback(
-        (metadataInstance: MetadataTemplateInstance) => {
-            const updatedInstances = templateInstances.filter(
-                templateInstance =>
-                    templateInstance.scope !== metadataInstance.scope &&
-                    templateInstance.templateKey !== metadataInstance.templateKey,
-            );
-            setTemplateInstances(updatedInstances);
-        },
-        [templateInstances],
-    );
-
     const handleDeleteMetadataInstance = React.useCallback(
-        (metadataInstance: MetadataTemplateInstance) => {
+        async (metadataInstance: MetadataTemplateInstance): Promise<void> => {
+            setStatus(STATUS.LOADING);
             if (!file || !metadataInstance) {
                 return;
             }
 
-            api.getMetadataAPI(false).deleteMetadata(
-                file,
-                metadataInstance,
-                deleteMetadataInstanceSuccessCallback,
-                onApiError,
-                true,
-            );
+            await api
+                .getMetadataAPI(false)
+                .deleteMetadata(file, metadataInstance, () => setStatus(STATUS.SUCCESS), onApiError, true);
         },
-        [api, onApiError, file, deleteMetadataInstanceSuccessCallback],
+        [api, onApiError, file],
     );
 
     const handleCreateMetadataInstance = React.useCallback(
-        (templateInstance: MetadataTemplateInstance, successCallback): void => {
-            api.getMetadataAPI(false).createMetadataRedesign(
-                file,
-                templateInstance,
-                successCallback,
-                (error: ElementsXhrError, code: string) =>
-                    onApiError(error, code, messages.sidebarMetadataEditingErrorContent),
-            );
+        async (templateInstance: MetadataTemplateInstance, successCallback: () => void): Promise<void> => {
+            await api
```

Comment:
Shouldn't we have `setStatus(STATUS.LOADING)` here too?

Reply:
For creating and updating we pass these async functions to `MetadataInstanceForm` in `metadata-editor` which is Formik form, and Formik should handle loading until it receives either a resolve or reject status from the API.